pytket/extensions/cutensornet/backends/cutensornet_backend.py

```diff
@@ -261,13 +259,46 @@ def get_operator_expectation_value(
         Returns:
             Expectation value.
         """
+        return self.get_matrix_element(
+            state_circuit,
+            state_circuit,
+            operator,
+            post_selection=post_selection,
+            valid_check=valid_check,
+        )
+
+    # TODO: this should be optionally parallelised with MPI
+    #  (both wrt Pauli strings and contraction itself).
+    def get_matrix_element(
+        self,
+        circuit_bra: Circuit,
+        circuit_ket: Circuit,
+        operator: QubitPauliOperator,
+        post_selection: Optional[dict[Qubit, int]] = None,
+        valid_check: bool = True,
+    ) -> float:
+        """Calculates a general matrix element using cuTensorNet contraction.
+
+        Has an option to do post selection on an ancilla register.
+
+        Args:
+            circuit_bra: Circuit representing bra state.
+            circuit_ket: Circuit representing ket state.
+            operator: Operator which matrix element is to be calculated.
+            valid_check: Whether to perform circuit validity check.
+            post_selection: Dictionary of qubits to post select where the key is
+                qubit and the value is bit outcome.
+
+        Returns:
+            Matrix element.
+        """
         if valid_check:
-            self._check_all_circuits([state_circuit])
+            self._check_all_circuits([circuit_bra, circuit_ket])
 
-        expectation = 0
+        element = 0
 
-        ket_network = TensorNetwork(state_circuit)
-        bra_network = ket_network.dagger()
+        bra_network = TensorNetwork(circuit_bra)
+        ket_network = TensorNetwork(circuit_ket).dagger()
```

Comment:
By convention, `bra` is the one that's daggered, instead of ket. Are you sure this is your intended one to dagger? Note that the order matters since `<a|O|b> = conj(<b|O|a>)`.

Reply:
sorry yes, obviously ... woops